### PR TITLE
add jpeg

### PR DESCRIPTION
**Pull request by gillins**
Unix build same as conda-recipes. 
Had to make some changes to the Windows build to make a DLL - this not supported by provided makefile so use simple cmake script instead that is provided as a patch with this recipe.

### CODE REVIEW

File: recipes/jpeg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        - vc9  # [win and py27]
+        - vc10  # [win and py34]
+        - vc14  # [win and py35]
+        
+requirements:
+    build:
+        - python  # [win]
+        - cmake  # [win]
+
+test:
```
Comment (patricksnape) [inline recipes/jpeg/meta.yaml]: As mentioned in the cross post, I don't think you need Python for building, but for testing - and you probably need to add a Jinja template to make sure the correct Python is chosen.

**Comment by patricksnape** [comment]
@msarahan Obviously Continuum already ship libjpeg - how are the Windows builds currently handled? Do you also use a custom CMake?

File: recipes/jpeg/meta.yaml (hunk 1 above)
Comment (gillins) [inline recipes/jpeg/meta.yaml]: OK I think I've got it working under Windows. Which is the Jinja template? appveyor seems to be running all the various versions of Python.

**Comment by gillins** [comment]
Continuum seem to use the makefile.vc that is supplied. However, this builds a static lib that can only be used by executables but not DLLs (ie GDAL). This workaround was the best I could come up with.

**Comment by patricksnape** [comment]
@gillins You are totally right, that is very confusing - because packages seem to be able to rely on jpeg and compile yet there is no DLL being created. And I've never set up a package to explicitly statically link jpeg so I'm stumped there. Great work on creating a working CMake for Windows then! I wonder if we need to test that it is exporting the symbols correctly somehow.

File: recipes/jpeg/meta.yaml (hunk 1 above)
Comment (patricksnape) [inline recipes/jpeg/meta.yaml]: The Jinja template is [here](https://github.com/conda-forge/staged-recipes/pull/358/files#diff-6db6ec9d7e8339bb0382b2c28399db45R25). That should get the builds working.

**Comment by patricksnape** [comment]
That failure is because of 503 errors :( Need to restart the build!

**Comment by gillins** [comment]
Well the Jinja thing seemed to have fixed it anyway since the Python 3.4 and 3.5 tests passed. Thanks!

Is restarting the build something I can do?

regarding symbols being exported under Windows: well the fact that the utility programs link seems a good test, but I could write some Python to test a list of function names are in the DLL using ctypes or win32.LoadLibrary if you think it is necessary.

**Comment by msarahan** [comment]
Sorry guys, I have been sick the past couple of days.  Confirming that the Continuum recipe only uses makefile.vc - so what you have here should be an improvement.  Wherever possible, it would be nice to provide both static and dynamic libraries, so anyone building with our stuff has the option.

**Comment by gillins** [comment]
Have added static libs for Unix and Windows builds. For Windows the dynamic libs are libjpeg.dll/.lib (the .lib is the import library) and the static lib is jpeg.lib. Not sure if there is a naming convention we should be following here.

Is it usual Conda convention to provide both static and dynamic versions? At one stage Ryan Grout seemed to be removing all the static builds.

AppVeyor seems happy now BTW.

**Comment by jakirkham** [comment]
> Is it usual Conda convention to provide both static and dynamic versions?

It is a conda-forge convention if that is what you mean, yes. We try to catch this and fix it when we can. If there are ones you know that are missing static libraries, feel free to raise an issue on the respective feedstock or note this in the PR to staged-recipes if it hasn't been released yet.

File: recipes/jpeg/build.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (jakirkham) [inline recipes/jpeg/build.sh]: Add `#!/bin/bash`

File: recipes/jpeg/build.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+./configure --prefix=$PREFIX \
+    --enable-shared=yes \
+    --enable-static=yes
+make
+make install
```
Comment (jakirkham) [inline recipes/jpeg/build.sh]: `make check`?

File: recipes/jpeg/build.sh (hunk 2 above)
Comment (jakirkham) [inline recipes/jpeg/build.sh]: Done.

**Comment by jakirkham** [comment]
This looks pretty nice to me. Thoughts from others?

We might want to squash the binary commits (pictures and such) so we don't have that stuff lingering in the repo's history.

**Comment by ocefpaf** [comment]
> This looks pretty nice to me. Thoughts from others?

:+1: 

> We might want to squash the binary commits (pictures and such) so we don't have that stuff lingering in the repo's history.

@gillins is in Australia, so I don't think anything will be done here in our time-zone. Maybe we could enable [squash-your-commits](https://github.com/blog/2141-squash-your-commits) :wink:

**Comment by jakirkham** [comment]
Ah, thanks for TZ info. Yeah, was thinking about doing it manually, but that is a good point.

**Comment by jakirkham** [comment]
Oh, squash merge is not enabled here. :open_mouth:

**Comment by jakirkham** [comment]
I'll do it the old fashion way.

**Comment by jakirkham** [comment]
Squashed and added in this PR ( https://github.com/conda-forge/staged-recipes/pull/369 ).

**Comment by ocefpaf** [comment]
Thanks @gillins for the recipe and @jakirkham for squashing it.

**Comment by gillins** [comment]
Yeah seems my TZ isn't very sociable... Thanks everyone for all the help getting this merged

**Comment by jakirkham** [comment]
No worries. It was easy enough to bring home. Thanks again @gillins.

**Comment by ocefpaf** [comment]
> Yeah seems my TZ isn't very sociable... Thanks everyone for all the help getting this merged 

Actually, for GitHub interactions, I do like that the PRs sit for a few hours :wink: 

However, it would be nice to get you in one of our Google Hangouts one day.

(See https://github.com/conda-forge/conda-forge.github.io/issues/89 for the last one.)

**Pull request closed without merging**